libobs/audio-monitoring/pulse/pulseaudio-output.c

```diff
@@ -0,0 +1,469 @@
+#include "obs-internal.h"
+#include "pulseaudio-wrapper.h"
+
+#define PULSE_DATA(voidptr) struct audio_monitor *data = voidptr;
+#define blog(level, msg, ...) blog(level, "pulse-am: " msg, ##__VA_ARGS__)
+
+struct audio_monitor {
+	obs_source_t *source;
+	pa_stream *stream;
+	char *device;
+
+	enum speaker_layout speakers;
+	pa_sample_format_t format;
+	uint_fast32_t samples_per_sec;
+	uint_fast32_t bytes_per_frame;
+	uint_fast8_t channels;
+
+	uint_fast32_t packets;
+	uint_fast64_t frames;
+
+	struct circlebuf new_data;
+	audio_resampler_t *resampler;
+	size_t buffer_size;
+	size_t bytesRemaining;
+
+	size_t bytes_per_channel;
+	bool ignore : 1;
```

Comment:
Consider using a regular boolean rather than a boolean bitfield.  There was a recent commit that converted boolean bitfields to regular booleans because the savings either weren't significant enough or the tradeoffs didn't make sense.  @notr1ch can probably elaborate if needed.